Service/ComponentenCatalogusService.php

```diff
@@ -381,17 +373,61 @@ public function importComponent($component): ?ObjectEntity
 
             return null;
         }
+        if (!$repositoryEntity = $this->getRepositoryEntity()) {
+            isset($this->io) && $this->io->error('No RepositoryEntity found when trying to import a Repository '.isset($component['name']) ? $component['name'] : '');
+
+            return null;
+        }
 
+        // Handle sync
         $synchronization = $this->synchronizationService->findSyncBySource($source, $componentEntity, $component['id']);
 
-        if (isset($this->io)) {
-            $this->io->comment('Mapping object'.$component['name']);
-            $this->io->comment('The mapping object '.$mapping);
+        isset($this->io) && $this->io->comment('Mapping object'.$component['name']);
+        isset($this->io) && $this->io->comment('The mapping object '.$mapping);
+
+        isset($this->io) && $this->io->comment('Checking component '.$component['name']);
+
+        // do the mapping of the component set two variables
+        $component = $componentArray = $this->mappingService->mapping($mapping, $component);
+        // unset component url before creating object, we don't want duplicate repositories
+        // we do not need the organisation set because this will be set in the FindOrganizationThroughRepositoriesService
```

Comment:
I do not fully understand what this comment on line 393 is about?
The comment is about an organisation, but we never do anything with a key or value called organisation?